demo/customchannel/main.js

```diff
@@ -76,3 +76,13 @@ friend.on('message', function (str) {
   page.emit('message', str);
 });
 
+page.on('req', function(url) {
+  page.emit('message', 'requiring ' + url);
+  core.require(url).then(function (Dependency) {
+    var d = new Dependency();
+    var method = '';
+    page.emit('message', 'has methods: ' + JSON.stringify(Object.keys(d)));
+  }, function (err) {
+    page.emit('message', 'Failed to load: ' + err);
+  });
+});
```

Comment:
one more thing to maintain. i like having an interactive page where I can easily run through different functionality interactively, which is why i added it here. it's a nicer workflow for me than the karma watch thing.